org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTDelegateCommandHandler.java

```diff
@@ -105,6 +105,12 @@ public Object executeCommand(String commandId, List<Object> arguments, IProgress
 					ProjectCommand.updateClasspaths(projectUri, entries.getClasspathEntries(), monitor);
 					return null;
 				}
+				case "java.project.updateProjectOptions": {

```

Comment:
I see you added the m2e profile setting to be retrievable under `getSettings`. Would you be open to calling the command `java.project.updateSettings` instead so it would be consistent with that existing command ?

Reply:
> Would you be open to calling the command java.project.updateSettings instead so it would be consistent with that existing command ?

Yes, this is better!